projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -364,12 +364,12 @@ function ( $parent_file ) use ( $menu_slug ) {
 	/**
 	 * Adds Appearance menu.
 	 *
-	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
+	 * @param String $customize_slug Whether customizer links should point to Calypso or wp-admin.
+	 * @param Bool   $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
 	 */
-	public function add_appearance_menu( $wp_admin = false ) {
+	public function add_appearance_menu( $customize_slug, $wp_admin = false ) {
```

Comment:
If you set `$customize_slug` as a class property defaulting to `customize.php`, you'd only need to update it in `Atomic_Admin_Menu`'s constructor. It would avoid this change and you wouldn't need to add new methods in the child classes.

Reply:
@obenland thanks for that suggestion! I have refactored the PR and it seems working for Simple, Atomic, Jetpack. I still have some problems with the tests though. I will need to look it further!